app/src/main/java/com/javahelp/frontend/activity/ProviderRegistrationViewmodel.java

```diff
@@ -1,17 +1,199 @@
+        this.username.setValue(username);
+    }
+
+    /**
+     * Sets the password for this {@link ProviderRegistrationViewmodel}
+     * @param password1 {@link String} password1 to use
+     */
+    public void setPassword1(String password1) {
+        this.password1.setValue(password1);
+    }
+
+    /**
+     * Sets the confirmed_password for this {@link ProviderRegistrationViewmodel}
+     * @param password2 {@link String} password2 to use
+     */
+    public void setPassword2(String password2) { this.password2.setValue(password2);}
+
+    /**
+     * Sets the email for this {@link ProviderRegistrationViewmodel}
+     * @param email {@link String} email to use
+     */
+    public void setEmail(String email) { this.email.setValue(email);}
+
+    /**
+     * Sets the firstname for this {@link ProviderRegistrationViewmodel}
+     * @param firstname {@link String} firstname to use
+     */
+    public void setFirstname(String firstname) { this.firstname.setValue(firstname);}
+
+    /**
+     * Sets the lastname for this {@link ProviderRegistrationViewmodel}
+     * @param lastname {@link String} lastname to use
+     */
+    public void setLastname(String lastname) { this.lastname.setValue(lastname);}
+
+    /**
+     * Sets the address for this {@link ProviderRegistrationViewmodel}
+     * @param address {@link String} address to use
+     */
+    public void setAddress(String address) { this.address.setValue(address);}
+
+    /**
+     * Sets the phone for this {@link ProviderRegistrationViewmodel}
+     * @param phone {@link String} phone to use
+     */
+    public void setPhone(String phone) { this.phone.setValue(phone);}
+
+    /**
+     * Sets whether the current provider is certified or not
+     * @param c whether certified
+     */
+    public void setCertified(boolean c) {certified.setValue(c);}
+
+    /**
+     * Sets the register error
+     * @param s {@link String} register error to set
+     */
+    public void setRegisterError(String s) {
+        registerResult.setValue(Optional.of(new RegisterResult(s)));}
+
+    /**
+     * Sets the provideruserinfo for this {@link ProviderRegistrationViewmodel}
+     * @param email, firstname, lastname, phone, address {@link String}
+     */
+    public void setProviderUserInfo(String email, String firstname, String lastname, String phone, String address){
+        this.providerUserInfo.setEmailAddress(email);
+        this.providerUserInfo.setPracticeName(firstname + lastname);
+        this.providerUserInfo.setCertified(false);
+        this.providerUserInfo.setAddress(address);
+        this.providerUserInfo.setPhoneNumber(phone);
+    }
+
+    /**
+     *
+     * @return the {@link RegisterResult} of the latest register attempt, if applicable
+     */
+    public MutableLiveData<Optional<RegisterResult>> getRegisterResult() {
+        return registerResult;
+    }
+
+
+    public boolean passwordMatch(String password1, String password2){
+        if (password1.equals(password2)){
```

Comment:
This could be further simplified into `return password1.equals(password2);`

Reply:
yes!